resources/setup.sh

```diff
@@ -139,27 +139,11 @@ chmod 644 /var/kube-config/*
 # fire after cluster hook, can be used for image pull / addon enabling whatevs
 source /after-cluster.sh
 
-# wait for pod start
-START_TIME=$(date +%s)
-while true; do
-    CURRENT_TIME=$(date +%s)
-    if [[ $((CURRENT_TIME-300)) -gt $START_TIME ]]; then
-        echo "Startup timeout, didn't become healthy after 2 mins.. details:"
-        /usr/local/bin/kubectl get po -n kube-system
-        exit 1
-    fi
-
-    echo "Checking startup status.."
-    POD_PHASES=$(/usr/local/bin/kubectl get po -n kube-system -o jsonpath='{.items[*].status.phase}' | tr ' ' '\n' | sort | uniq)
-    POD_STATES=$(/usr/local/bin/kubectl get po -n kube-system -o jsonpath='{.items[*].status.containerStatuses[*].state}' | tr ' ' '\n' | cut -d'[' -f 2 | cut -d':' -f 1 | sed 's|["{}]*||g' | sort | uniq)
-    POD_READINESS=$(/usr/local/bin/kubectl get po -n kube-system -o jsonpath='{.items[*].status.containerStatuses[*].ready}' | tr ' ' '\n' | sort | uniq)
-    POD_COUNT=$(/usr/local/bin/kubectl get po -n kube-system --no-headers | wc -l)
-    if [ "$POD_READINESS" == "true" ] && [ "$POD_STATES" == "running" ] && [ "$POD_PHASES" == "Running" ] && [ $POD_COUNT -ge 7 ]; then
-        echo "startup successful"
-        break
-    fi
-    sleep 5
-done
+if ! kubectl wait --for=condition=ready --timeout 5m pod --all --all-namespaces; then
```

Comment:
Why `--all-namespaces` and not just `kube-system`?

Reply:
`wait` (https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#wait) is experimental and known to be a bit flaky.

Reply:
I manually verified that all version up to v1.19.4 only has pods in kube-system and `--all-namespaces` was more a catch all for future release

Reply:
I did not know that `wait` was experimental but glad that I know it now. If we dont want to use wait the only option that I could think is do it the old way with complex queries that deals with multiple versions of K8S

Reply:
dunno the `wait` approach is pretty nice. I've not found it flakey? ive used it elsewhere, do you think it would be an issue @furikake ?

Reply:
Mine was around running right after an apply. You need to give the scheduler enough time so that wait will evaluate everything or you'll miss some.

BTW. You know that if you have 10 pods here then `wait` will actually wait for 50 mins.

Quick proof:

```
❯ kgp --all-namespaces| wc -l
      18

❯ gtime kubectl wait --for=condition=succeeded --timeout 2s pod --all --all-namespaces
timed out waiting for the condition on pods/nginx-deployment-575fc7645b-2nb85
timed out waiting for the condition on pods/nginx-deployment-575fc7645b-6mtkp
timed out waiting for the condition on pods/nginx-deployment-575fc7645b-8s87m
timed out waiting for the condition on pods/nginx-deployment-575fc7645b-gh5pq
timed out waiting for the condition on pods/nginx-deployment-575fc7645b-lzcl6
timed out waiting for the condition on pods/nginx-deployment-575fc7645b-ng9b4
timed out waiting for the condition on pods/nginx-deployment-575fc7645b-pckb5
timed out waiting for the condition on pods/nginx-deployment-575fc7645b-v8hr5
timed out waiting for the condition on pods/coredns-76d76cbc76-5r2k4
timed out waiting for the condition on pods/coredns-76d76cbc76-ft96g
timed out waiting for the condition on pods/etcd-docker-desktop
timed out waiting for the condition on pods/kube-apiserver-docker-desktop
timed out waiting for the condition on pods/kube-controller-manager-docker-desktop
timed out waiting for the condition on pods/kube-proxy-jmn9n
timed out waiting for the condition on pods/kube-scheduler-docker-desktop
timed out waiting for the condition on pods/storage-provisioner
timed out waiting for the condition on pods/vpnkit-controller
Command exited with non-zero status 1
0.14user 0.06system 0:34.46elapsed 0%CPU (0avgtext+0avgdata 30132maxresident)k
0inputs+0outputs (10major+7717minor)pagefaults 0swaps
``` 

17 pods * 2 seconds = 34 seconds

Reply:
@itsnagaraj maybe just drop the timeout value a bit (like 2m) and that's probably good enough as you don't want it to wait for too long. I'm assuming there's enough time between scheduling and waiting so my race condition isn't a problem.

Maybe you can look into whether `--watch` is a better fit in the future? 

```
❯ kgp --watch -o=custom-columns='NAME:metadata.name, STATUS:.status.phase' --no-headers -n kube-system
coredns-76d76cbc76-5r2k4                 Running
coredns-76d76cbc76-ft96g                 Running
etcd-docker-desktop                      Running
kube-apiserver-docker-desktop            Running
kube-controller-manager-docker-desktop   Running
kube-proxy-jmn9n                         Running
kube-scheduler-docker-desktop            Running
storage-provisioner                      Running
vpnkit-controller                        Running
```

Reply:
@furikake I originally started without timeout but version > v1.19.0 are failing to startup within default wait timeout of 30s. I will reduce the timeout bit to 2m and finger crossed should not have any issues

Reply:
@furikake setup.sh script is for building the image and starting the cluster and committing the docker images and not for actually starting the kubernetes cluster for testing purposes. I am thinking of leaving the wait time to 5 minutes so what is committed as docker image is actually started up. Do you still see this as an issue?